File: Java/InverterString.java
package Java;
import java.util.Scanner;

public class InverterString {
    public static String inverterString(String s) {
        char[] caracteres = s.toCharArray();
        int tamanho = caracteres.length;

        for (int i = 0; i < tamanho / 2; i++) {
            char temp = caracteres[i];
            caracteres[i] = caracteres[tamanho - i - 1];
            caracteres[tamanho - i - 1] = temp;
        }

        return new String(caracteres);
    }

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.print("Digite uma string para inverter: ");
            String stringOriginal = scanner.nextLine();

            String stringInvertida = inverterString(stringOriginal);

            System.out.println("String Original: " + stringOriginal);
            System.out.println("String Invertida: " + stringInvertida);
        }
    }
}

File: Python/InverterString.py
def inverter_string(s):
    caracteres = list(s)

    tamanho = len(caracteres)

    for i in range(tamanho // 2):
        temp = caracteres[i]
        caracteres[i] = caracteres[tamanho - i - 1]
        caracteres[tamanho - i - 1] = temp

    string_invertida = ''.join(caracteres)

    return string_invertida

string_original = input("Digite uma string para inverter: ")
string_invertida = inverter_string(string_original)

print("String Original:", string_original)
print("String Invertida:", string_invertida)
